tensorflow_quantum/datasets/spin_system.py

```diff
@@ -312,6 +312,248 @@ def tfi_chain(qubits, boundary_condition="closed", data_dir=None):
+    circuit = cirq.Circuit(cirq.X.on_each(qubits))
+    circuit.append(cirq.H(qubits[i]) for i in range(0, nspins, 2))
+    circuit.append(
+        cirq.CNOT(qubits[j], qubits[j + 1]) for j in range(0, nspins - 1, 2))
+
+    odd_qubits = list(
+        zip([qubits[j] for j in range(1, nspins - 1, 2)],
+            [qubits[j + 1] for j in range(1, nspins - 1, 2)]))
+    even_qubits = list(
+        zip([qubits[j] for j in range(0, nspins - 1, 2)],
+            [qubits[j + 1] for j in range(0, nspins - 1, 2)]))
+    for d in range(depth):
+        for q1, q2 in odd_qubits:
+            circuit.append(cirq.ZZ(q1, q2)**(symbols[d]))
+            circuit.append(cirq.YY(q1, q2)**(symbols[d + depth]))
+            circuit.append(cirq.XX(q1, q2)**(symbols[d + depth]))
+        if boundary_condition == "closed":
+            circuit.append(cirq.ZZ(qubits[-1], qubits[0])**(symbols[d]))
+            circuit.append(cirq.YY(qubits[-1], qubits[0])**(symbols[d + depth]))
+            circuit.append(cirq.XX(qubits[-1], qubits[0])**(symbols[d + depth]))
+        for q1, q2 in even_qubits:
+            circuit.append(cirq.ZZ(q1, q2)**(symbols[d + 2 * depth]))
+            circuit.append(cirq.YY(q1, q2)**(symbols[d + 3 * depth]))
+            circuit.append(cirq.XX(q1, q2)**(symbols[d + 3 * depth]))
+    # Initiate lists.
+    resolved_circuits = []
+    hamiltonians = []
+    order_parameters = []
+    additional_info = []
+    labels = []
+    # Load the data and append to the lists.
+    for i, directory in enumerate(x for x in os.listdir(data_path)):
+        # The folders are named according to the order value data they contain.
+        g = float(directory)
+        with open(os.path.join(data_path, directory, "stats.txt"), "r") as file:
+            lines = file.readlines()
+            res_e = float(lines[0].split("=")[1].strip("\n"))
+            fidelity = float(lines[2].split("=")[1].strip("\n"))
+        order_parameters.append(g)
+        params = np.load(os.path.join(data_path, directory, "params.npy")) \
+                 / np.pi
+        params = params.astype(np.float64)
+        additional_info.append(
+            SpinSystemInfo(g=g,
+                           gs=np.load(
+                               os.path.join(data_path, directory,
+                                            "groundstate.npy"))[:, 0],
+                           gs_energy=np.load(
+                               os.path.join(data_path, directory,
+                                            "energy.npy"))[0],
+                           res_energy=res_e,
+                           fidelity=fidelity,
+                           params=dict(zip(symbol_names, params.flatten())),
+                           var_circuit=circuit))
+        # Resolve the circuit parameters.
+        param_resolver = cirq.resolve_parameters(circuit,
+                                                 additional_info[i].params)
+        resolved_circuits.append(param_resolver)
```

Comment:
Would you mind referencing your Cirq Bug in a comment in the code so we know to revisit it later and why there is a cast of numeric types ?